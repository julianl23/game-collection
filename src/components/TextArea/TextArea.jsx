import React, { Component } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const StyledTextArea = styled.textarea`
  display: block;
  border: 1px solid ${({ theme }) => theme.borderGray};
  background: ${({ theme }) => theme.purpleWhite};
  font-size: ${({ theme }) => theme.fontSizeNormal};
  padding: 15px 10px;
  height: 90px;
  width: 100%;
`;

StyledTextArea.displayName = "TextArea";

const Label = styled.label`
  display: block;
  margin-bottom: 10px;
`;

const Wrapper = styled.div`
  margin-bottom: 20px;
`;

const ErrorMessage = styled.p`
  color: ${({ theme }) => theme.pinkRed};
  font-size: 1em;
  font-weight: 500;
  margin: 10px 0 20px;
`;

class TextArea extends Component {
  handleChange = event => {
    const { onChange } = this.props;
    onChange(event);
  };

  handleBlur = event => {
    const { onBlur } = this.props;
    onBlur(event);
  };

  render() {
    const { id, value, name, className, labelText, errorMessage } = this.props;

    return (
      <Wrapper className={className}>
        {labelText && <Label htmlFor={id}>{labelText}</Label>}
        <StyledTextArea
          id={id}
          value={value}
          name={name}
          onChange={this.handleChange}
          onBlur={this.handleBlur}
        />
        {errorMessage && (
          <div id={`${id}-error-message`} className="textfield-error-message">
            <ErrorMessage>{errorMessage}</ErrorMessage>
          </div>
        )}
      </Wrapper>
    );
  }
}

TextArea.propTypes = {
  id: PropTypes.string.isRequired,
  className: PropTypes.string,
  labelText: PropTypes.string,
  errorMessage: PropTypes.string,
  name: PropTypes.string.isRequired,
  value: PropTypes.string,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
};

TextArea.defaultProps = {
  value: null,
  className: "",
  labelText: "",
  errorMessage: "",
  onChange: () => {},
  onBlur: () => {},
};

export default TextArea;
