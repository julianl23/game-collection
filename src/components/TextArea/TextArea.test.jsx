import React from "react";
import TestRenderer from "react-test-renderer";
import { shallow } from "enzyme";

import TextArea from "./TextArea";

describe("TextArea", () => {
  it("renders correctly", () => {
    const instance = TestRenderer.create(<TextArea id="hello" />);
    expect(instance.toJSON()).toMatchSnapshot();
  });

  it("displays the passed in value when one is set", () => {
    const instance = TestRenderer.create(<TextArea id="hello" value="world" />);
    expect(instance.toJSON()).toMatchSnapshot();
  });

  it("displays a label if labelText is set", () => {
    const wrapper = shallow(
      <TextArea id="hello" labelText="Hello" value="world" />
    );

    expect(
      wrapper
        .find("Label")
        .first()
        .text()
    ).toEqual("Hello");
  });

  it("displays an error message if errorMessage is set", () => {
    const wrapper = shallow(
      <TextArea id="hello" errorMessage="You done a bad" value="world" />
    );

    expect(
      wrapper
        .find("ErrorMessage")
        .first()
        .text()
    ).toEqual("You done a bad");
  });
});
