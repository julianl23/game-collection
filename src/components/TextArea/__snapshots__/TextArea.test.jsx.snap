// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea displays the passed in value when one is set 1`] = `
.c1 {
  display: block;
  border: 1px solid;
  padding: 15px 10px;
  height: 90px;
  width: 100%;
}

.c0 {
  margin-bottom: 20px;
}

<div
  className="c0"
>
  
  <textarea
    className="c1"
    id="hello"
    onBlur={[Function]}
    onChange={[Function]}
    value="world"
  />
  
</div>
`;

exports[`TextArea renders correctly 1`] = `
.c1 {
  display: block;
  border: 1px solid;
  padding: 15px 10px;
  height: 90px;
  width: 100%;
}

.c0 {
  margin-bottom: 20px;
}

<div
  className="c0"
>
  
  <textarea
    className="c1"
    id="hello"
    onBlur={[Function]}
    onChange={[Function]}
    value={null}
  />
  
</div>
`;
